frontend/app/src/composables/history/events/index.ts

```diff
@@ -144,32 +154,80 @@ export const useTransactions = () => {
     }
   };
 
-  const fetchTransactions = async (
-    payload: MaybeRef<TransactionRequestPayload>
-  ): Promise<Collection<EthTransactionEntry>> => {
-    const result = await fetchEthTransactions({
-      ...get(payload),
-      onlyCache: true
-    });
-    const mapped = mapCollectionEntriesWithMeta<EthTransactionEntry>(
-      mapCollectionResponse(result)
-    );
+  const reDecodeMissingTransactionEventsTask = async (
+    account: EvmChainAddress
+  ) => {
+    const taskType = TaskType.TX_EVENTS;
+
+    const payload: AddressesAndEvmChainPayload = {
+      evmChain: account.evmChain,
+      addresses: [account.address]
+    };
 
-    startPromise(
-      Promise.allSettled([
-        fetchEnsNames(getNotesAddresses(mapped.data)),
-        fetchTransactionEvents(
-          mapped.data.filter(
-            ({ decodedEvents }) => decodedEvents && decodedEvents.length === 0
-          )
-        )
-      ])
+    if (get(isTaskRunning(taskType, payload))) {
+      return;
+    }
+
+    try {
+      const { taskId } = await reDecodeMissingTransactionEvents<PendingTask>([
+        payload
+      ]);
+
+      const taskMeta = {
+        title: t('actions.transactions_events.task.title').toString(),
+        description: tc(
+          'actions.transactions_events.task.description',
+          2,
+          account
+        ),
+        numericKeys: [],
```

Comment:
do we still use numeric keys anywhere?

Reply:
no I think, this one is leftover, will remove